embed/etcd.go

```diff
@@ -343,6 +350,10 @@ func (e *Etcd) serve() (err error) {
 }
 
 func (e *Etcd) errHandler(err error) {
+	if transport.IsClosedConnError(err) || err == grpc.ErrServerStopped {
```

Comment:
```go
// in embed/etcd.go
func StartEtcd(inCfg *Config) (e *Etcd, err error) {
	if err = inCfg.Validate(); err != nil {
		return nil, err
	}
	e = &Etcd{cfg: *inCfg, stopc: make(chan struct{})}
	cfg := &e.cfg
	defer func() {
		if e != nil && err != nil {
			e.Close()
			e = nil
		}
	}()
```

We close `stopc` by calling `e.Close()` here, but `StartEtcd` returns with `nil` error, so it is not called in our use case?